toolchain/lexer/string_literal.cpp

```diff
@@ -53,12 +69,14 @@ auto LexedStringLiteral::Lex(llvm::StringRef source_text)
   llvm::SmallString<16> terminator("\"");
   llvm::SmallString<16> escape("\\");
 
-  const int multi_line_prefix_size =
-      GetMultiLineStringLiteralPrefixSize(source_text.substr(hash_level));
-  const bool multi_line = multi_line_prefix_size > 0;
+  const int multi_line_prefix_size = GetMultiLineStringLiteralPrefixSize(
+      source_text.substr(hash_level), terminator);
+  const MultiLineKind multi_line = multi_line_prefix_size == 0 ? NotMultiLine
+                                   : terminator[0] == '"'
+                                       ? MultiLineWithDoubleQuotes
+                                       : MultiLine;
```

Comment:
I feel like this is getting to be a bit hard to follow with the densely encoded information across the API boundary. What do you think about a simpler approach of returning a tuple of the kind and size?

Reply:
Thanks for the suggestion, I ended up with a named class rather than a tuple; I think this is substantially clearer.